src/Database/LSMTree/Internal/BlobFile.hs

```diff
@@ -64,12 +71,14 @@ removeReference BlobFile{blobFileRefCounter} =
 newBlobFile ::
      PrimMonad m
   => HasFS m h
+  -> RemoveFileOnClose
   -> FS.Handle h
   -> m (BlobFile m h)
-newBlobFile fs blobFileHandle = do
+newBlobFile fs r blobFileHandle = do
     let finaliser = do
           FS.hClose fs blobFileHandle
-          FS.removeFile fs (FS.handlePath blobFileHandle)
+          unless (r == DoNotRemoveFileOnClose) $
+            FS.removeFile fs (FS.handlePath blobFileHandle)
```

Comment:
A simpler hack/TODO would be

```suggestion
          -- TODO: uncomment the next line when we have properly implemented snapshotting
          -- FS.removeFile fs (FS.handlePath blobFileHandle)
```

and then there is no need to modify code in other places than right here

Reply:
The write buffer blobs code _relies_ on deleting the file. Tests fail if we don't do it. Hence the hacky do/don't flag.